feat(readonlyvalue): add loading state

erzhan-temir-mamyrov opened the pull request:
Closes #3655 

**Summary**

- Add prop for loading state

**Change List (commits, features, bugs, etc)**

- Add `isLoading` prop to component
- Add `skeletonLoadingLabel` and `skeletonLoadingValue` for configuration
- Update stories, add new story `with async data load`
- Update `publicAPI`
- Update docs

**Acceptance Test (how to verify the PR)**

- Go to [ReadOnlyValue / multiple values - stacked](https://deploy-preview-3679--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-readonlyvalue--multiple-values-stacked)
- In the knob tab "Other" enable `isLoading`
- Verify loading state

</br>

- Go to [ReadOnlyValue / multiple values - inline](https://deploy-preview-3679--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-readonlyvalue--multiple-values-inline)
- In the knob tab "Other" enable `isLoading`
- Verify loading state

</br>

- Go to [ReadOnlyValue / with async data load](https://deploy-preview-3679--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-readonlyvalue--with-async-data-load)
- Verify loading state and transition to values

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

<!-- For reviewers: do not remove -->

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

JordanWSmith15 reviewed (changes requested):
1. Once label02 and value02 load into view, they are appearing much lower than where the skeleton was positioned. This does not happen with label/value 01
2. We should expose a property to control the width of the skeletons. Having them fill 100% of their container width makes sense as a default, but may make a page very "busy" if there are a lot of these, particularly in a single, wide column.

erzhan-temir-mamyrov commented:
@JordanWSmith15 Thanks for the feedback! PR is updated:
- The height of a value depends upon the content to be loaded. I added knob `Line count` to control the height of value during loading state.
- I added storybook knob to control width for label and value.

JordanWSmith15 commented:
@davidicus looks good on my end
